.github/workflows/publish.yml

```diff
@@ -0,0 +1,62 @@
+# Version tags only
+
+name: Publish
+
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+
+env:
+  NODE_VERSION: 18
+
+jobs:
+  npmPublish:
+    name: Publish to npm
+    runs-on: ubuntu-latest
+
+    env:
+      # It'll work with secrets.GITHUB_TOKEN (which is provided by GitHub unconditionally)
+      # Still then release author would be "github-actions". It's better if it's dedicated repo bot
+      GITHUB_TOKEN: ${{ secrets.USER_GITHUB_TOKEN }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Retrieve dependencies from cache
+        id: cacheNpm
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.npm
+            node_modules
+          key: npm-v${{ env.NODE_VERSION }}-${{ runner.os }}-${{ github.ref }}-${{ hashFiles('package.json') }}
+          restore-keys: |
+            npm-v${{ env.NODE_VERSION }}-${{ runner.os }}-${{ github.ref }}-
+            npm-v${{ env.NODE_VERSION }}-${{ runner.os }}-refs/heads/master-
```

Comment:
As we're in tag context I bekieve it should be just:

```
key: npm-v${{ env.NODE_VERSION }}-${{ runner.os }}-refs/heads/master-${{ hashFiles('package.json') }}
```

(no `restore-keys`)

Reply:
removed restore keys, good catch!